Cargo.toml

```diff
@@ -24,7 +24,7 @@ lazy_static = "0.2"
 time = "0.1"
 hyper = "0.10"
 getopts = "0.2"
-protobuf = "1.1"
+protobuf = "2.2"
```

Comment:
We've had issues with bumping protobuf before (i.e. deleting all the data). So I'd rather not bump this if it's not required.

Reply:
I tried compiling with the protobuf version in master and it spit out a bunch of errors about `Cell<u32>` being unsafe to share between threads.

Reply:
Well, I just tried this code on an existing database (backup) and it deleted it all.

So it'd be important to figure out how to implement this feature without bumping the protobuf version!

Reply:
I feel like we should have some sort of database format that's more robust to change. Bumping a version shouldn't delete the entire thing.

@ergl Do you have any ideas on how to do this with the current protobuf version?

Reply:
There's plans to make this less volatile.

Reply:
Hi! I played around with storing a "version header" byte along with the raw protocol buffer binary on disk, that way we could at least know if we're dealing with an old file.

It's certainly troubling that a library bump would delete the database file, and that's the root of this problem, it would be nice to know why that is.

Reply:
Considering that the storage backend is meant to be refactored in a few different ways, I would probably opt to reject this PR for now, as much as it nice to have the feature.

It's definitely something to keep track of (is there an issue?) but there's other work to get out the way for now before we can have a look at changing the storage format

Reply:
Seems fair, definitely having it tracked (issue, github project, or whatever) would be perfect. If you need any help, I'd be down to contribute.

Reply:
Feel free to open an issue. I'll create a few milestones to tag stuff with